eml/screens/explore/Explore.jsx

```diff
@@ -1,87 +1,81 @@
-/* eslint-disable camelcase */
-import React, { useEffect, useState } from 'react'
-import { View, Text, Platform, ScrollView } from 'react-native'
-import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
-import { AppLoading } from 'expo-app-loading'
-import { SelectList } from 'react-native-dropdown-select-list'
-import ActiveExploreCard from '../../components/explore/ActiveExploreCard'
-import ExploreCard from '../../components/explore/ExploreCard'
-import * as StorageService from '../../services/StorageService'
-import { useIsFocused } from '@react-navigation/native'
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { View, Text, Image } from 'react-native';
+import FilterNavBar from '../../components/explore/FilterNavBar';
+import { ScrollView } from 'react-native-gesture-handler';
+import ExploreCard from '../../components/explore/ExploreCard';
+import { client } from '../../api/userApi';
+import {getCourseList} from '../../services/StorageService';
+import { getSubCourseList } from '../../services/StorageService';
 
 
-export default function Explore() {
 
-    const isFocused = useIsFocused()
+import { getBucketImage } from "../../api/api";
 
-    const [selected, setSelected] = useState(-1);
+function Explore() {
+  
+  // Search text state
+  const [searchText, setSearchText] = useState('');
+  // Selected category state
+  const [selectedCategory, setSelectedCategory] = useState(null);
 
-    const [nonActiveViews, setViews] = useState([]);
-    const [activeViews, setViews2] = useState([]);
-    // All, Sustainability, Finance, Other
-    const uniqueCategories = [{ key: -1, value: "Todos" }, { key: '635f9ae2991d8c6da796a1cc', value: "Sustentabilidade" }, { key: '6368be5d71e079ae8d537eb1', value: "Finança" }, { key: '639208a0f467689fde25b5fa', value: "De outros" }];
+  //Sets dummy data for courses (will be replaced with data from backend)
+  const [courses, setCourses] = useState([
 
-    const [fontsLoaded] = useFonts({
-        VarelaRound_400Regular
-    })
+]);
 
-    async function loadViews() {
-        const courseList = await StorageService.getCourseList();
-        let componentPromises = courseList.map(({ title, iconPath, isActive, courseId, categoryId }, index) => {
-            if (isActive === false) {
-                if ((isActive && categoryId === selected) || (isActive && selected === -1)) {
-                    return <ActiveExploreCard key={index} title={title} courseId={courseId} iconPath={iconPath} />;
-                } else if ((!(isActive) && categoryId === selected) || (!(isActive) && selected === -1)) {
-                    return <ExploreCard key={index} title={title} courseId={courseId}></ExploreCard>
-                }
-            }
-        });
-        Promise.all(componentPromises).then(setViews);
+//Fetch courses from backend and replace dummy data!
+useEffect(() => {
```

Comment:
My earlier comment about creating an infinite loop also applies here if correct. I can't test it right now, as I get a fatal error when entering the exercises page on this branch.

Reply:
This is also true